### ISSUE TITLE
Folder Type granularity in sharing

### ISSUE THREAD

**Issue by Kedryn** [syncthing/syncthing#9419]
I think it would be useful to choose if a folder is Sent/receive/both, at sharing level.
I meant when, in a folder detail, we choose to who share it. For example, i could share to some friends the folder in send_only, but with one that collaborates in manintaining it, in send_receive
Right now we can decide only at folder level, se we have to share it two times

**Comment by tomasz1986** [comment]
I think this is an interesting idea, however I cannot imagine how the current Web GUI would even be able to support this. If you've got anything concrete in mind, I believe that it would be useful to have at least a brief sketch of how this could possibly look visually in the GUI.

**Comment by er-pa** [comment]
It is an interesting idea..but the GUI is probably the least of the hurdles here;p The folder type is quite a fundamental part of Syncthing. Changing that to a per-device configuration is probably going to be a very tricky move. Which, of course, doesn't mean that it can't happen or that it's not going to happen - just that it'll take quite some time and effort.

As long as you have all the nodes in control, this is already possible. I'm personally not sure the effort is worth the benefit here. But that's just my take. Perhaps I'm just missing the actual potential of this:)

**Comment by Kedryn** [comment]
what do you mean with "As long as you have all the nodes in control, this is already possible."?

**Comment by er-pa** [comment]
> what do you mean with "As long as you have all the nodes in control, this is already possible."?

What I meant is that as long as you can rely on the Syncthing Config files staying as expected, under your control, you can set your 'friends' locations' to receive only and practically have the situation you aim for.

**Comment by Kedryn** [comment]
> I think this is an interesting idea, however I cannot imagine how the current Web GUI would even be able to support this. If you've got anything concrete in mind, I believe that it would be useful to present at least a brief sketch of how this could possibly look visually in the GUI.

I'm awful at sketches, but i was thinking at something like this

![image](https://github.com/syncthing/syncthing/assets/12759632/9be96263-15ed-4c2e-8f71-7888e86c797c)

That would allow me (as others) to de-dupe a lot of folders, scans, update checks, and so on

**Comment by Kedryn** [comment]
> It is an interesting idea..but the GUI is probably the least of the hurdles here;p The folder type is quite a fundamental part of Syncthing. Changing that to a per-device configuration is probably going to be a very tricky move. Which, of course, doesn't mean that it can't happen or that it's not going to happen - just that it'll take quite some time and effort.

I'm a system manager, i quite agree

> As long as you have all the nodes in control, this is already possible. I'm personally not sure the effort is worth the benefit here. But that's just my take. Perhaps I'm just missing the actual potential of this:)

mainly reduce the dupes of shared folders, rescans, gui cluttering (i have a lot of shares, duped for distribution only or collaboration)

**Comment by tomasz1986** [comment]
> I'm awful at sketches, but i was thinking at something like this
> 
> ![image](https://private-user-images.githubusercontent.com/12759632/305060406-9be96263-15ed-4c2e-8f71-7888e86c797c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMDU2MjMsIm5iZiI6MTcwODAwNTMyMywicGF0aCI6Ii8xMjc1OTYzMi8zMDUwNjA0MDYtOWJlOTYyNjMtMTVlZC00YzJlLThmNzEtNzg4OGU4NmM3OTdjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE1VDEzNTUyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwYzFiMTdkMDhkNDM1NTM3YmQ2ZTk5NWU3YjVmODBkNWE0MjFiZjhiMDdkOGE3YTgxMThkZDNiYmNkMjNjMWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gqazMUhFrH24sqQYgUyMN5dSSu4epoiVb-Mt1N9EZGU)
> 
> That would allow me (as others) to de-dupe a lot of folders, scans, update checks, and so on

I think one big problem would be that currently, depending on the folder type, the folder state changes, e.g. you get "Local Additions" and the "Override Local Changes" button for Receive Only folders, the "Override Changes" button for Send Only folders, etc. If a folder can have different types, those states don't make much sense any more.

**Comment by luca1234567a** [comment]
The architecture of syncthing that the Folder Type was only on for the Folder may create security isues when sync different machine with different admin.
I think a solution on Local/Remote Changes can be found.
I think we may change terminology, every folder have more device type. 
We consider a device with a share with all type of other devices.
For the device with Receive Only, we may have the problem of local change, so may be the "Override Local Changes" button.
For the device with Send Only, we may have the problem of remote change, so may be the "Override Remote Changes" button. For semplicity may be consider all the device with "Send Only" with the same states.
For the device with Send&Receive Only, I think may there is not problem with change.

I problem became when two device with sharing togher, have all two setting Send Only or Receive Only.

**Comment by tdussa** [comment]
> > what do you mean with "As long as you have all the nodes in control, this is already possible."?
> 
> What I meant is that as long as you can rely on the Syncthing Config files staying as expected, under your control, you can set your 'friends' locations' to receive only and practically have the situation you aim for.

That is true, but in practice, this is also wishful thinking IMHO.  This is almost never truly the case, and even if we propose that no involved user will ever change their config maliciously, all it takes is a successful hacker attack or an involuntary misconfiguration on some unsuspecting user's part.  While this of course will work fine 99 % of the time, in my opinion, it would really be nice to have the possibility to make _sure_ that nothing bad will happen. :)
